includes/Service/YoutubeVanisher.php

```diff
@@ -76,9 +149,68 @@ protected function getVideoData($markup) {
     $data = parent::getVideoData($markup);
     $data['video_id'] = $this->extractVideoId($data['src']);
 
+    // Extract additional parameters from URL or iframe attributes.
+    $data = array_merge($data, $this->getUrlParams($data['src']));
+    $data = array_merge($data, $this->getIframeAttributes($markup));
+
     return $data;
   }
 
+  /**
+   * Extracts URL parameters from the YouTube URL.
+   *
+   * @param string $url
+   *   The YouTube URL.
+   *
+   * @return array
+   *   Array of parameters.
+   */
+  protected function getUrlParams($url) {
+    $params = [];
+    $parsed_url = parse_url($url);
+
+    if (isset($parsed_url['query'])) {
+      parse_str($parsed_url['query'], $query_params);
+
+      // Parameters to preserve
+      foreach (self::PRESERVED_PARAMS as $param) {
+        if (isset($query_params[$param])) {
+          $params[$param] = $query_params[$param];
+        }
+      }
+
+      // Convert 't' to 'start' if exists.
+      if (isset($query_params['t'])) {
+        $t_value = preg_replace('/s$/i', '', $query_params['t']);
+        $params['start'] = $t_value;
+      }
+    }
+
+    return $params;
+  }
+
+  /**
+   * Extracts attributes from the iframe element.
+   *
+   * @param string $markup
+   *   The iframe markup.
+   *
+   * @return array
+   *   Array of attributes.
+   */
+  protected function getIframeAttributes($markup) {
+    $attributes = [];
+
+    foreach (self::PRESERVED_PARAMS as $attr) {
+      $pattern = '/' . $attr . '=["\']([^"\']*)["\']|' . $attr . '=([^\s>]*)/i';
+      if (preg_match($pattern, $markup, $matches)) {
+        $attributes[$attr] = !empty($matches[1]) ? $matches[1] : $matches[2];
+      }
+    }
+
+    return $attributes;
+  }
```

Comment:
Sorry, I wasn't completely clear.  It also makes sense to say 'attribute' in full.

```suggestion
  protected function getIframeAttributes($markup) {
    $attributes = [];

    foreach (self::PRESERVED_PARAMS as $attribute) {
      $pattern = '/' . $attribute . '=["\']([^"\']*)["\']|' . $attribute . '=([^\s>]*)/i';
      if (preg_match($pattern, $markup, $matches)) {
        $attributes[$attribute] = !empty($matches[1]) ? $matches[1] : $matches[2];
      }
    }

    return $attributes;
  }
```